### PR TITLE
arch/risc-v: Move xxx_cpustart.c to common

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
It's a common solution for MSIP and IPI based risc-v smp soc,
also works on qemu-rv smp (WIP).
## Impact
None
## Testing
CI and maix-bit.

### CODE REVIEW

File: arch/risc-v/src/common/riscv_cpustart.c [hunk 1]
```diff
@@ -41,37 +41,24 @@
...
  ****************************************************************************/
 
 extern volatile bool g_serial_ok;
-extern int riscv_pause_handler(int irq, void *c, void *arg);
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
+extern int riscv_pause_handler(int irq, void *c, void *arg);
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_cpustart.c:58]: should we move to riscv_internal.h?

Comment (no1wudi) [reply arch/risc-v/src/common/riscv_cpustart.c]: Done

File: arch/risc-v/src/common/riscv_cpustart.c [hunk 2]
```diff
@@ -41,37 +41,22 @@
...
-#  define showprogress(c) riscv_lowputc(c)
-#else
-#  define showprogress(c)
-#endif
-
 /****************************************************************************
  * Public Data
  ****************************************************************************/
 
 extern volatile bool g_serial_ok;
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_cpustart.c:52]: remove g_serial_ok too?

Comment (no1wudi) [reply arch/risc-v/src/common/riscv_cpustart.c]: Ok, and also removed it from non-smp system.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

File: arch/risc-v/src/common/riscv_internal.h [hunk 3]
```diff
@@ -116,6 +116,10 @@ EXTERN uint32_t g_intstackalloc; /* Allocated stack base */
 EXTERN uint32_t g_intstacktop;   /* Initial top of interrupt stack */
 #endif
 
+#ifdef CONFIG_SMP
+EXTERN volatile bool g_serial_ok;
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/riscv_internal.h:120]: only k210 really use g_serial_ok, so it's better to move it to k210 specific header file instead.

Comment (no1wudi) [reply arch/risc-v/src/common/riscv_internal.h:120]: Yes, only k210 need it now, but other smp system also need it, qemu-rv smp support is ready to submit.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**